packages/google_sign_in/google_sign_in_platform_interface/lib/google_sign_in_platform_interface.dart

```diff
@@ -89,6 +92,40 @@ abstract class GoogleSignInPlatform {
     throw UnimplementedError('init() has not been implemented.');
   }
 
+  /// Initializes the plugin. You must call this method before calling other
+  /// methods.
+  ///
+  /// The [hostedDomain] argument specifies a hosted domain restriction. By
+  /// setting this, sign in will be restricted to accounts of the user in the
+  /// specified domain. By default, the list of accounts will not be restricted.
+  ///
+  /// The list of [scopes] are OAuth scope codes to request when signing in.
+  /// These scope codes will determine the level of data access that is granted
+  /// to your application by the user. The full list of available scopes can be
+  /// found here: <https://developers.google.com/identity/protocols/googlescopes>
+  ///
+  /// The [signInOption] determines the user experience. [SigninOption.games] is
+  /// only supported on Android.
+  ///
+  /// The [forceCodeForRefreshToken] is used on Android to ensure the authentication
+  /// code can be exchanged for a refresh token after the first request.
+  ///
+  /// See:
+  /// https://developers.google.com/identity/sign-in/web/reference#gapiauth2initparams
+  Future<void> initWithForceCodeForRefreshToken({
+    List<String> scopes = const <String>[],
+    SignInOption signInOption = SignInOption.standard,
+    String? hostedDomain,
+    String? clientId,
+    bool forceCodeForRefreshToken = false,
```

Comment:
I should have flagged this before but missed it: rather than add the specific parameter, we should add a parameter object to future-proof this. So:
```
Future<void> initWithParams(SignInInitParameters params);
```

where that's a new class that contains all the current parameters. Then next time, we can just add a new parameter there instead of making a new method.

(No need to re-review here with that change, it'll get formal review sign-off in the platform interface PR.)